core/LSU.cpp

```diff
@@ -596,6 +608,36 @@ namespace olympia
         sparta_assert(false, "Attempt to complete instruction no longer exiting in issue queue!");
     }
 
+    // Pop completed load instruction out of load queue
+    void LSU::popLoadQueue_(const InstPtr & inst_ptr)
+    {
+        // Look for the instruction to be completed, and remove it from load queue
+        for (auto iter = load_queue_.begin(); iter != load_queue_.end(); iter++) {
+            if ((*iter)->getInstPtr() == inst_ptr) {
+                load_queue_.erase(iter);
+
+                return;
+            }
+        }
```

Comment:
A faster way to do this: when the load is allocated into the load queue, store the index (iterator) of its position in the request.  Then, when popping, check that iterator's validity and if valid, just erase it.